src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplate.java

```diff
@@ -242,13 +242,14 @@ private void populateDetainedFields(AsylumCase asylumCase, Map<String, Object> f
                 fieldValues.put("nomsAvailable", nomsAvailable);
 
                 YesOrNo releaseDateProvided = YesOrNo.NO;
-                String prisonerReleaseDate = formatComplexString(asylumCase
-                        .get("dateCustodialSentence")
-                        .toString());
-
-                if (!prisonerReleaseDate.isBlank()) {
-                    releaseDateProvided = YesOrNo.YES;
-                    fieldValues.put("releaseDate", prisonerReleaseDate);
+                if (asylumCase.containsKey("dateCustodialSentence")) {
```

Comment:
instead containsKey, we use read method on FieldDefinitions. Can we use that?

Reply:
I believe, No. Reason being, dateCustodialSentence, prisonerNOMSNumber & otherDetentionFacilityName are not sent as simple strings to the server. Therefore they need to be manually extracted from asylumCase and parsed to remove braces/brackets etc.

Reply:
This is a complex type field. Hence we cant use the read method.